nvflare/private/fed/server/server_engine.py

```diff
@@ -693,13 +549,10 @@ def persist_components(self, fl_ctx: FLContext, completed: bool):
 
             job_info = fl_ctx.get_prop(FLContextKey.JOB_INFO)
             if not job_info:
-                with self.parent_conn_lock:
-                    data = {ServerCommandKey.COMMAND: ServerCommandNames.GET_CLIENTS, ServerCommandKey.DATA: {}}
-                    self.parent_conn.send(data)
-                    return_data = self.parent_conn.recv()
-                    job_id = return_data.get(ServerCommandKey.JOB_ID)
-                    job_clients = return_data.get(ServerCommandKey.CLIENTS)
-                    fl_ctx.set_prop(FLContextKey.JOB_INFO, (job_id, job_clients))
+                return_data = self._get_clients_data(job_id)
+                job_id = return_data.get(ServerCommandKey.JOB_ID)
+                job_clients = return_data.get(ServerCommandKey.CLIENTS)
+                fl_ctx.set_prop(FLContextKey.JOB_INFO, (job_id, job_clients))
             else:
                 (job_id, job_clients) = job_info
             snapshot.set_component_snapshot(
```

Comment:
do we need the `self.parent_conn_lock` anymore?

Reply:
no need any more. Removed.